lib/easypost/beta/beta_referral.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+class EasyPost::BetaReferral < EasyPost::Resource
+  # Create a referral user. This function requires the Partner User's API key.
+  def self.create(params = {}, api_key = nil)
+    response = EasyPost.make_request(:post, '/beta/referral_customers', api_key, { user: params })
+    EasyPost::Util.convert_to_easypost_object(response, api_key)
+  end
+
+  # Update a referral user. This function requires the Partner User's API key.
+  def self.update_email(email, user_id, api_key = nil)
+    wrapped_params = {
+      :user => {
+        :email => email,
+      }
+    }
+    _ = EasyPost.make_request(:put, "/beta/referral_customers/#{user_id}", api_key, wrapped_params)
+
+    # return true if API succeeds, else an error is throw if it fails.
+    true
+  end
+
+  # Retrieve a list of referral users. This function requires the Partner User's API key.
+  def self.all(params = {}, api_key = nil)
+    response = EasyPost.make_request(:get, '/beta/referral_customers', api_key, params)
+    EasyPost::Util.convert_to_easypost_object(response, api_key)
+  end
+
+  # Add credit card to a referral user. This function requires the Referral User's API key.
+  def self.add_credit_card(referral_api_key, number, expiration_month, expiration_year, cvc, priority = 'primary')
+    easypost_stripe_api_key = retrieve_easypost_stripe_api_key
+
+    begin
+      stripe_credit_card_token = create_stripe_token(
+        number,
+        expiration_month,
+        expiration_year,
+        cvc,
+        easypost_stripe_api_key,
+        )
+    rescue Exception => e
+      raise EasyPost::Error.new('Could not send card details to Stripe, please try again later.')
+    end
+
+    response = create_easypost_credit_card(referral_api_key, stripe_credit_card_token, priority)
+    EasyPost::Util.convert_to_easypost_object(response, referral_api_key)
```

Comment:
Does this work properly? We don't yet define a `CreditCard` object in this lib

Reply:
Yes, it will convert to `EasyPost::EasyPostObject`